lib/CGameState.cpp

```diff
@@ -1090,19 +1090,6 @@ void CGameState::placeCampaignHeroes()
 			logGlobal->debug("\tReplace placeholders with heroes");
 			replaceHeroesPlaceholders(campaignHeroReplacements);
 
-			// remove hero placeholders on map
-			for(auto obj : map->objects)
```

Comment:
Just in case somebody decided to put hero placeholder on regular map moved this from if